manager/controllers/app/blueprint_controller.go

```diff
@@ -51,41 +52,87 @@ func (r *BlueprintReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	if err := r.Get(ctx, req.NamespacedName, &blueprint); err != nil {
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
-	observedStatus := blueprint.Status.DeepCopy()
+	if err := r.reconcileFinalizers(&blueprint); err != nil {
+		log.V(0).Info("Could not reconcile finalizers " + err.Error())
+		return ctrl.Result{}, err
+	}
 
-	if blueprint.DeletionTimestamp != nil {
+	// If the object has a scheduled deletion time, update status and return
+	if !blueprint.DeletionTimestamp.IsZero() {
+		// The object is being deleted
 		log.V(0).Info("Reconcile: Deleting Blueprint " + blueprint.GetName())
-	} else {
-		log.V(0).Info("Reconcile: Installing/Updating Blueprint " + blueprint.GetName())
+		return ctrl.Result{}, nil
 	}
 
+	observedStatus := blueprint.Status.DeepCopy()
+	log.V(0).Info("Reconcile: Installing/Updating Blueprint " + blueprint.GetName())
+
 	result, err := r.reconcile(ctx, log, &blueprint)
 	if err != nil {
 		return ctrl.Result{}, errors.Wrap(err, "failed to reconcile blueprint")
 	}
 
-	if blueprint.DeletionTimestamp == nil {
-		if !equality.Semantic.DeepEqual(&blueprint.Status, observedStatus) {
-			if err := r.Client.Status().Update(ctx, &blueprint); err != nil {
-				return ctrl.Result{}, errors.WrapWithDetails(err, "failed to update blueprint status", "status", blueprint.Status)
-			}
+	if !equality.Semantic.DeepEqual(&blueprint.Status, observedStatus) {
+		if err := r.Client.Status().Update(ctx, &blueprint); err != nil {
+			return ctrl.Result{}, errors.WrapWithDetails(err, "failed to update blueprint status", "status", blueprint.Status)
 		}
 	}
 
 	log.Info("blueprint reconcile cycle completed", "result", result)
 	return result, nil
 }
 
-func (r *BlueprintReconciler) deleteChartResource(ctx context.Context, kubeNamespace string, releaseName string) (ctrl.Result, error) {
-	rel, err := r.Helmer.Status(kubeNamespace, releaseName)
-	if err == nil && rel != nil {
-		_, _ = r.Helmer.Uninstall(kubeNamespace, releaseName)
+// reconcileFinalizers reconciles finalizers for Blueprint
+func (r *BlueprintReconciler) reconcileFinalizers(blueprint *app.Blueprint) error {
+	// finalizer
+	finalizerName := r.Name + ".finalizer"
+	hasFinalizer := utils.ContainsFinalizer(blueprint.GetFinalizers(), finalizerName)
```

Comment:
I believe that we can use `ctrlutil.ContainsFinalizer` now (we use v0.6.2 of controller-runtime)

https://github.com/kubernetes-sigs/controller-runtime/blob/v0.6.2/pkg/controller/controllerutil/controllerutil.go#L290

Reply:
Done.